content/docs/pipelines/reference/component-spec.md

```diff
@@ -54,17 +54,10 @@ This section describes the
 * `description`: Description of the component.
 * `metadata`: Standard object's metadata:
 
-    * `annotations`: An unstructured key value map
-        stored with a resource. External tools can set this property to store and
-        retrieve arbitrary metadata. Annotations are not queryable and should be
-        preserved when modifying objects. See more information in the
+    * `annotations`: A string key-value map used to add information about the component. Current limitation: the key cannot contain more that one slash ("/").
+        The annotations get translated to Kubernetes annotations when the component task is executed. See more information in the
         [Kubernetes user guide](http://kubernetes.io/docs/user-guide/annotations).
 
-    * `labels`: A map of string keys and values that can be used to
```

Comment:
Are labels no longer supported? I see they're still in the spec. As a reader, it's a bit confusing to see the `metadata` list item with only one sub-item (`annotations`). If for some reason we don't want to describe labels, we should either just mention them briefly as a second list item, or change the `annotations` entry from a list item to a continuation of the `metadata` list item.

Reply:
`labels` are still supported, but I want to de-emphasize them and possibly deprecate in future.
When I was adding annotations and labels I did not consider them to be Kubernetes-specific, but now I see them as such (especially labels). The component spec should be portable, so Kubernetes-specific attributes are note desirable.
If we forget about the Kubernetes-related properties of those two attributes, then the become essentially the same.
So the idea is to at least deprecate the `labels` as too Kubernetes-specific in favor of `annotations`.
I can list `labels` as "deprecated".

>As a reader, it's a bit confusing to see the metadata list item with only one sub-item

The outline follows the structure of the component specification file:

```yaml
metadata:
    annotations:
        k1: v1
        k2: v2
    labels:
        k1: v1
        k2: v2
```

Reply:
I've added `labels` back with a deprecation notice.

Reply:
Kubernetes adopters use labels extensively. They are not the same as annotations. If you merge them into annotations, you

a) make things confusing for users
b) will break Kubernetes tools that people like us have built, e.g. monitoring and accounting

Reply:
> Kubernetes adopters use labels extensively. They are not the same as annotations.

I know about Kubernetes labels and annotations.
The annotations in Kubeflow Pipelines component specifications are not the same as Kubernetes annotations though. They were never intended to be the same as Kubernetes' annotations/labels. They were copied to Kubernetes pods at one point, but it was a mistake. Especially for labels (since they are very Kubernetes-specific). All platform-specific options should be specified on component instances, not on components themselves.

>will break Kubernetes tools that people like us have built, e.g. monitoring and accounting

Can you please tell me more about cases where you use Kubeflow Pipelines Components with Kubernetes tools.

Reply:
We do resource accounting and troubleshooting based on Kubernetes labels. We use our own pipeline tools right now, but we are starting to test and will eventually adopt KF Pipelines. We need to find a way to propagate labels to the pods that run in the cluster. If you don't want to keep both, perhaps deprecate annotations instead? Labels are a more powerful construct, since that's what Kubernetes selectors allow you to use. Prometheus can tag data with either annotations or labels from pods, but e.g. Stackdriver Monitoring, DataDog and SignalFx only look at labels.

So right now the plan is for people to call `add_pod_label()` manually after instantiating a component? That means the burden is now shifted to all the callers, correct?